library/std/src/io/error.rs

```diff
@@ -141,6 +141,16 @@ struct Custom {
 /// It is used with the [`io::Error`] type.
 ///
 /// [`io::Error`]: Error
+///
+/// # Handling errors and matching on `ErrorKind`
+///
+/// In application code, use `match` for the `ErrorKind` values you are expecting; use `_` to match
+/// "all other errors".
+///
+/// In comprehensive and thorough tests that want to verify that a test doesn't return any known incorrect error kind,
+/// you may want to cut-and-paste the current list of errors from here into your test code. This seems counterintuitive,
```

Comment:
```suggestion
/// you may want to cut-and-paste the current full list of errors from here into your test code, and then match `_` as the correct case. This seems counterintuitive,
```

(modulo word wrapping)